rootston/output.c

```diff
@@ -512,6 +532,26 @@ static void render_output(struct roots_output *output) {
 
 damage_finish:
 	pixman_region32_fini(&damage);
+
+	// Send frame done events to all surfaces
+	if (output->fullscreen_view != NULL) {
+		view_for_each_surface(view, surface_send_frame_done, &data);
```

Comment:
There should be a view = output->fullscreen_view before that?

Also, is that correct if the fullscreen view isn't fully opaque? How do we handle semi-transparent fullscreen stuff?

Reply:
The output->fullscreen_view effectivly disables compositing.

If you check the render code, it skips rendering anything else as well.

Reply:
It does? That seems wrong, what if the fullscreen view is semitransparent?

Reply:
The spec says fullscreen views are rendered on a black background.

Reply:
The spec can get stuffed, then.

Reply:
@martinetd Also note that I originally didn't paid attention to send frame done events to surfaces only once, because doing so flushes the clients callbacks. But yeah, it's better to do it right.